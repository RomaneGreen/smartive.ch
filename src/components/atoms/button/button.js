import React from 'react';
import PropTypes from 'prop-types';
import Link from 'gatsby-link';

import './button.scss';

const mod = modifier => ` button--${modifier}`;

export const Button = ({ url, text, isPrimary, isWhite, hasBorder }) => {
  const internal = /^\/(?!\/)/.test(url);
  const className = `button${isPrimary ? mod('primary') : ''}${isWhite ? mod('white') : ''}${hasBorder ? ' has-border' : ''}`;
  if (internal) {
    return (
      <Link to={url} className={className}>
        {text}
      </Link>
    );
  }
  return (
    <a href={url} className={className}>
      {text}
    </a>
  );
};

export default Button;

Button.propTypes = {
  url: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  isPrimary: PropTypes.bool,
  isWhite: PropTypes.bool,
  hasBorder: PropTypes.bool,
};

Button.defaultProps = {
  isPrimary: false,
  isWhite: false,
  hasBorder: false,
};
