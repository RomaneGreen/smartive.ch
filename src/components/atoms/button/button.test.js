import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import { Button } from './button';

// This test uses the Shallow Renderer (only one JSX level deep) because the
// underlying gatsby-link relies on the router being passed in the context.
// Therefore this test doesn't test the link implementation.
const renderer = new ShallowRenderer();

describe('Button', () => {
  it('renders default external link correctly', () => {
    const tree = renderer.render(<Button url="http://example.com" text="Button" />);
    expect(tree).toMatchSnapshot();
  });

  it('renders default internal link correctly', () => {
    const tree = renderer.render(<Button url="/internal" text="Button" />);
    expect(tree).toMatchSnapshot();
  });

  it('renders primary correctly', () => {
    const tree = renderer.render(<Button url="http://example.com" text="Primary" isPrimary />);
    expect(tree).toMatchSnapshot();
  });

  it('renders white with border correctly', () => {
    const tree = renderer.render(<Button url="http://example.com" text="White w/ Border" isWhite hasBorder />);
    expect(tree).toMatchSnapshot();
  });
});
