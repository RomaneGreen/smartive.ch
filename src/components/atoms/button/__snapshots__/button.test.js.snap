// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders default external link correctly 1`] = `
<a
  className="button"
  href="http://example.com"
>
  Button
</a>
`;

exports[`Button renders default internal link correctly 1`] = `
<GatsbyLink
  className="button"
  to="/internal"
>
  Button
</GatsbyLink>
`;

exports[`Button renders primary correctly 1`] = `
<a
  className="button button--primary"
  href="http://example.com"
>
  Primary
</a>
`;

exports[`Button renders white with border correctly 1`] = `
<a
  className="button button--white has-border"
  href="http://example.com"
>
  White w/ Border
</a>
`;
